lib/datadog/tracing/contrib/grpc/datadog_interceptor/client.rb

```diff
@@ -44,6 +44,9 @@ def annotate!(trace, span, metadata, call)
               host, _port = find_host_port(call)
               span.set_tag(Tracing::Metadata::Ext::TAG_PEER_HOSTNAME, host) if host
 
+              deadline = find_deadline(call)
+              span.set_tag(Ext::TAG_CLIENT_DEADLINE, (deadline.to_time.to_f * 1000).to_i) if deadline
```

Comment:
I'm not sure `to_time` is needed here. It seems like the object is always a `Time` object.

Also, `Time#to_time` is not part of the Ruby standard library:
```sh
$ ruby -e 'Time.at(1).to_time'
-e:1:in `<main>': undefined method `to_time' for 1969-12-31 16:00:01 -0800:Time (NoMethodError)
```
It works in our tests because it seems like some gem is monkey patching `Time`, but we can't guarantee it.

Reply:
@marcotc it's actually a `GRPC::Core::TimeSpec` object, not a `Time` object

Reply:
![Screenshot from 2022-08-04 07-32-35](https://user-images.githubusercontent.com/1973623/182738391-2bc6443e-cf1d-4f81-baa2-cb72d4168c49.png)

Reply:
But, you're right, there is some cases, deadline is a `Time` object, and ActiveSupport is monkey patching `Time`
![Screenshot from 2022-08-04 07-47-20](https://user-images.githubusercontent.com/1973623/182739462-0656fb75-ee43-406a-bceb-1d53f3311a88.png)

let me address it

Reply:
Turns out that gRPC gem mentioned that issue in their spec https://github.com/grpc/grpc/blob/3d2e5038c310df3b4d215cb7872a26045e979547/src/ruby/spec/generic/client_stub_spec.rb#L79-L83

So the value of the deadline can be Time or TimeSpec either. That's why I handle 2 cases in this commithttps://github.com/DataDog/dd-trace-rb/pull/2200/commits/f7499de35d6d3dd9fad652cadd385da7ab331fea